### ISSUE TITLE
Stemcell: btrfs_delayed_qgroup_accounting crashes 

### ISSUE THREAD

**Issue by combor** [cloudfoundry/bosh#1345]
Starting from version 1.3 Concourse uses btrfs for container image management. This leads to kernel errors like this one:

```
Jul  8 13:09:58 localhost kernel: [10217.120526] ------------[ cut here ]------------
Jul  8 13:09:58 localhost kernel: [10217.120557] WARNING: CPU: 1 PID: 6588 at /build/linux-lts-vivid-TukaE5/linux-lts-vivid-3.19.0/fs/btrfs/qgroup.c:1414 btrfs_delayed_qgroup_accounting+0x6ef/0xf00 [btrfs]()
Jul  8 13:09:58 localhost kernel: [10217.120559] Modules linked in: xt_addrtype nf_log_ipv4 nf_log_common xt_LOG ipt_MASQUERADE nf_nat_masquerade_ipv4 veth bridge stp llc btrfs xor raid6_pq iptable_nat nf_nat_ipv4 nf_nat ipt_REJECT nf_reject_ipv4 nf_conntrack_ipv4 nf_defrag_ipv4 xt_conntrack nf_conntrack iptable_filter ip_tables x_tables aufs intel_rapl iosf_mbi cirrus ttm crct10dif_pclmul crc32_pclmul drm_kms_helper drm xen_fbfront fb_sys_fops syscopyarea aesni_intel ppdev sysfillrect aes_x86_64 sysimgblt lrw gf128mul 8250_fintek glue_helper i2c_piix4 ablk_helper parport_pc cryptd mac_hid parport serio_raw nfsd auth_rpcgss nfs_acl nfs lockd grace sunrpc fscache psmouse pata_acpi ixgbevf(OE) floppy
Jul  8 13:09:58 localhost kernel: [10217.120602] CPU: 1 PID: 6588 Comm: docker Tainted: G           OE  3.19.0-64-generic #72~14.04.1-Ubuntu
Jul  8 13:09:58 localhost kernel: [10217.120604] Hardware name: Xen HVM domU, BIOS 4.2.amazon 05/12/2016
Jul  8 13:09:58 localhost kernel: [10217.120606]  0000000000000000 ffff880024a839c8 ffffffff817b563f 0000000000000000
Jul  8 13:09:58 localhost kernel: [10217.120609]  ffffffffc063eaf8 ffff880024a83a08 ffffffff81075d0a 0000000000000000
Jul  8 13:09:58 localhost kernel: [10217.120612]  ffff88000fdf7458 ffff880144a60be0 ffff880106975808 0000000000000000
Jul  8 13:09:58 localhost kernel: [10217.120615] Call Trace:
Jul  8 13:09:58 localhost kernel: [10217.120623]  [<ffffffff817b563f>] dump_stack+0x63/0x81
Jul  8 13:09:58 localhost kernel: [10217.120627]  [<ffffffff81075d0a>] warn_slowpath_common+0x8a/0xc0
Jul  8 13:09:58 localhost kernel: [10217.120630]  [<ffffffff81075dfa>] warn_slowpath_null+0x1a/0x20
Jul  8 13:09:58 localhost kernel: [10217.120644]  [<ffffffffc061de3f>] btrfs_delayed_qgroup_accounting+0x6ef/0xf00 [btrfs]
Jul  8 13:09:58 localhost kernel: [10217.120656]  [<ffffffffc05a61aa>] btrfs_run_delayed_refs.part.66+0x20a/0x270 [btrfs]
Jul  8 13:09:58 localhost kernel: [10217.120668]  [<ffffffffc05a6227>] btrfs_run_delayed_refs+0x17/0x20 [btrfs]
Jul  8 13:09:58 localhost kernel: [10217.120681]  [<ffffffffc05b6ff7>] btrfs_commit_transaction+0x47/0xa50 [btrfs]
Jul  8 13:09:58 localhost kernel: [10217.120695]  [<ffffffffc05eb8d7>] btrfs_mksubvol.isra.28+0x537/0x550 [btrfs]
Jul  8 13:09:58 localhost kernel: [10217.120699]  [<ffffffff810b6350>] ? prepare_to_wait_event+0x110/0x110
Jul  8 13:09:58 localhost kernel: [10217.120711]  [<ffffffffc05eba7f>] btrfs_ioctl_snap_create_transid+0x18f/0x1a0 [btrfs]
Jul  8 13:09:58 localhost kernel: [10217.120723]  [<ffffffffc05ebbe6>] btrfs_ioctl_snap_create_v2+0xe6/0x140 [btrfs]
Jul  8 13:09:58 localhost kernel: [10217.120735]  [<ffffffffc05ef6f6>] btrfs_ioctl+0xfd6/0x2ae0 [btrfs]
Jul  8 13:09:58 localhost kernel: [10217.120739]  [<ffffffff811fe033>] ? path_openat+0x93/0x5c0
Jul  8 13:09:58 localhost kernel: [10217.120743]  [<ffffffff81062d8a>] ? __do_page_fault+0x1ea/0x5a0
Jul  8 13:09:58 localhost kernel: [10217.120746]  [<ffffffff81201d78>] do_vfs_ioctl+0x2f8/0x510
Jul  8 13:09:58 localhost kernel: [10217.120750]  [<ffffffff81117074>] ? __audit_syscall_entry+0xb4/0x110
Jul  8 13:09:58 localhost kernel: [10217.120753]  [<ffffffff81023a8c>] ? do_audit_syscall_entry+0x6c/0x70
Jul  8 13:09:58 localhost kernel: [10217.120755]  [<ffffffff81202011>] SyS_ioctl+0x81/0xa0
Jul  8 13:09:58 localhost kernel: [10217.120759]  [<ffffffff817bd14d>] system_call_fastpath+0x16/0x1b
Jul  8 13:09:58 localhost kernel: [10217.120761] ---[ end trace 14515625064c1faa ]---
```

It is highly likely that it is related to `btrfs_delayed_qgroup_accounting` crash that was reported here: http://www.mail-archive.com/linux-btrfs%40vger.kernel.org/msg42197.html.

I'm using `3262.2` stemcell.

**Comment by cppforlife** [comment]
@combor was this resolved in 4.4 kernel (3263.x+ stemcells)?

**Comment by mfine30** [comment]
Closing due to inactivity
